app.rb

```diff
@@ -0,0 +1,179 @@
+require './book'
+require './person'
+require './decorator'
+require './rental'
+require './student'
+require './teacher'
+require './classroom'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def run
+    puts 'Welcome to School Library App!'
+    loop do
+      interface_menu
+      option = gets.chomp
+      if option == '7'
+        puts 'Thank you for using the Library App!'
+        break
+      end
+      option_selector(option)
+    end
+  end
+
+  def list_books
+    @books.each { |book| puts "#{book.title} by #{book.author}" }
+  end
+
+  def list_people
+    @people.each { |person| puts "#{person.name} - #{person.age}" }
+  end
+
+  def list_rentals_person
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index { |person, index| puts "#{index}) #{person.name}" }
+    person_index = gets.chomp.to_i
+    show_rentals_person(person_index)
+  end
+
+  def show_rentals_person(person_index)
+    puts "#{@people[person_index].name} has rented the following books:"
+    @people[person_index].rentals.map do |rental|
+      puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}"
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]'
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'That is not a valid input'
+      nil
+    end
+  end
+
+  def create_student
+    id = @people.length + 1
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Has parent permission? [Y/N]?'
+    permission = gets.chomp
+    if permission.match?(/y/i)
+      parent_permission = true
+    elsif permission.match?(/n/i)
+      parent_permission = false
+    else
+      puts 'That is not a valid input'
+      return
+    end
+    student = Student.new(id, age, name, parent_permission)
+    add_person(student)
+  end
+
+  def create_teacher
+    id = @people.length + 1
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Age:'
+    age = gets.chomp
+    puts 'Specialization:'
+    specialization = gets.chomp
+    teacher = Teacher.new(id, age, name, specialization)
```

Comment:
:red_circle: Please make sure you arrange the parameters properly. If you look at the constructor definition in your `teacher.rb` file the first argument is the specialization, the second is age, and the last one is name. Since the parent_permission is always true for the teachers' object there is no need to include the parent_permission parameter. Your constructor should look like this :arrow_down: 

```
def initialize(specialization, age, name = 'Unknown')
    super(name, age )
    @specialization = specialization
  end
```

You should now create an instance as follows :arrow_down: 

` Teacher.new(specialization, age, name)`